### PR TITLE
[fix][doc] Add doc for `ackReceiptEnabled` config

### LINKED ISSUE #16449: [Doc] Add missing consumer config "ackReceiptEnabled" in the Pulsar document
#### Actual behavior
In the current Pulsar doc for Java client consumer configuration, the parameter "ackReceiptEnabled" is missing
https://pulsar.apache.org/docs/next/client-libraries-java/#configure-consumer

### PR DESCRIPTION
**Pull request by coderzc**
Fixes #16449 

### Motivation

Missing consumer config `ackReceiptEnabled`.

### Modifications

Add doc for `ackReceiptEnabled` config.

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by RobertIndie — APPROVED** [review]

File: site2/docs/client-libraries-java.md [hunk 1]
```diff
@@ -743,6 +743,7 @@ When you create a consumer, you can use the `loadConf` configuration. The follow
 `autoAckOldestChunkedMessageOnQueueFull`|boolean|Whether to automatically acknowledge pending chunked messages when the threashold of `maxPendingChunkedMessage` is reached. If set to `false`, these messages will be redelivered by their broker. |true
 `maxPendingChunkedMessage`|int| The maximum size of a queue holding pending chunked messages. When the threshold is reached, the consumer drops pending messages to optimize memory utilization.|10
 `expireTimeOfIncompleteChunkedMessageMillis`|long|The time interval to expire incomplete chunks if a consumer fails to receive all the chunks in the specified time period. The default value is 1 minute. | 60000
+`ackReceiptEnabled`|boolean| If `ackReceiptEnabled` is enabled, ACK will return receipt but does not mean that the message will not be resent after get receipt.
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-java.md:746]:
```suggestion
`ackReceiptEnabled`|boolean| If `ackReceiptEnabled` is enabled, ACK returns a receipt. The message is resent after getting the receipt.
```

Write in the simple present tense as much as possible if you are covering facts that were, are, and forever shall be true. https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.e8uqh1awkcnp

Comment (coderzc) [reply site2/docs/client-libraries-java.md]: OK, thanks

**Review by dave2wave — APPROVED** [review]
LGTM

File: site2/docs/client-libraries-java.md [hunk 2]
```diff
@@ -743,6 +743,7 @@ When you create a consumer, you can use the `loadConf` configuration. The follow
 `autoAckOldestChunkedMessageOnQueueFull`|boolean|Whether to automatically acknowledge pending chunked messages when the threashold of `maxPendingChunkedMessage` is reached. If set to `false`, these messages will be redelivered by their broker. |true
 `maxPendingChunkedMessage`|int| The maximum size of a queue holding pending chunked messages. When the threshold is reached, the consumer drops pending messages to optimize memory utilization.|10
 `expireTimeOfIncompleteChunkedMessageMillis`|long|The time interval to expire incomplete chunks if a consumer fails to receive all the chunks in the specified time period. The default value is 1 minute. | 60000
+`ackReceiptEnabled`|boolean| If `ackReceiptEnabled` is enabled, ACK returns a receipt. The message is resent after getting the receipt.
```
Comment (michaeljmarshall) [inline site2/docs/client-libraries-java.md:746]: > The message is resent after getting the receipt.

@coderzc - can you explain what this is supposed to mean? My understanding of `ackReceipt` is simply that the broker responds to consumer ack commands with an `ackResponse`.

File: site2/docs/client-libraries-java.md (hunk 1 above)
Comment (michaeljmarshall) [reply site2/docs/client-libraries-java.md]: We'll need to revisit this doc. It's not correct to say `The message is resent after getting the receipt.`.

Comment (lhotari) [reply site2/docs/client-libraries-java.md]: > We'll need to revisit this doc. It's not correct to say `The message is resent after getting the receipt.`.

@coderzc @Anonymitaet Please follow up on @michaeljmarshall 's comment above. /cc @eolivelli

Comment (Anonymitaet) [reply site2/docs/client-libraries-java.md]: Thanks for your suggestions. Feel free to correct docs directly.

File: site2/docs/client-libraries-java.md (hunk 2 above)
Comment (coderzc) [reply site2/docs/client-libraries-java.md:746]: You are right, I will fix it.

File: site2/docs/client-libraries-java.md (hunk 1 above)
Comment (coderzc) [reply site2/docs/client-libraries-java.md]: I fixed it in PR: https://github.com/apache/pulsar/pull/16923

**Pull request merged**